### PR TITLE
Query parameters getter for URL

### PR DESCRIPTION
**Pull request by nathanbacon**
This is useful for iMessage app/extensions because information is exchanged through URL query items but it might be useful in general.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a changelog entry describing my changes.

### CODE REVIEW

File: Tests/FoundationTests/URLExtensionsTests.swift [hunk 1]
```diff
@@ -33,5 +33,17 @@ final class URLExtensionsTests: XCTestCase {
 		url.appendQueryParameters(params)
 		XCTAssertEqual(url, queryUrl)
 	}
+    
+    func testQueryItemsVar() {
```
Comment (SD10) [inline Tests/FoundationTests/URLExtensionsTests.swift]: Can you name this `testQueryParameters`? We use the naming scheme of `test` + method or property name.

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 2]
```diff
@@ -8,6 +8,23 @@
 
 import Foundation
 
+// MARK: - Properties
+public extension URL {
+    
+    /// SwifterSwift: Dictionary of the URL's query parameters
+    public var queryParameters: [String: String?]? {
```
Comment (SD10) [inline Sources/Extensions/Foundation/URLExtensions.swift]: Can't the key just be of type `String`? Looking up a value in a dictionary returns an optional value by default

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,6 +40,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `random(from: Date, upTo: Date) -> Date` method that return radom date in in the specified range [#336](https://github.com/SwifterSwift/SwifterSwift/pull/336) by [akuzminskyi](https://github.com/akuzminskyi).
   - Added `string(withFormat format: String)` method to get a string from a date with the given format.
   - Added `init?(integerLiteral value: Int)` initializer to create date object from Int literal. [#342](https://github.com/SwifterSwift/SwifterSwift/pull/342) by [n0an](https://github.com/n0an).
+- **URL**
+  - Added `queryParmeters` property to get the query parameters from a URL as a dictionary by [nathanbacon](https://github.com/nathanbacon).
```
Comment (SD10) [inline CHANGELOG.md]: Please include the PR number in the CHANGELOG. Use the previous entries as a reference

**Comment by nathanbacon** [comment]
@SD10 I addressed each point you had. It now returns a `[String: String]` and an unassigned query parameter is just an empty string and I update the test for this case.

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 4]
```diff
@@ -8,6 +8,23 @@
...
+public extension URL {
+    
+    /// SwifterSwift: Dictionary of the URL's query parameters
+    public var queryParameters: [String: String]? {
+        guard let components = URLComponents(url: self, resolvingAgainstBaseURL: false), let queryItems = components.queryItems else { return nil }
+        
+        var items: [String: String] = [:]
+        
+        for queryItem in queryItems {
+            items[queryItem.name] = queryItem.value ?? ""
```
Comment (SD10) [inline Sources/Extensions/Foundation/URLExtensions.swift]: any reason why we set an empty String? shouldn't it just be nil?

Comment (SD10) [reply Sources/Extensions/Foundation/URLExtensions.swift]: cc @nathanbacon if you make this change I can get this merged in 👍

Comment (nathanbacon) [reply Sources/Extensions/Foundation/URLExtensions.swift]: @SD10, nil makes more sense. Changed this.

**Comment by SD10** [comment]
Thanks for updating this @nathanbacon. I'm going to hold off merging until I setup Aeryn

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Pull request merged**